acl2017/acl2017.github.io
# Adding alpha version of program for testing.

This branch provides a version of the program page that allows users to generate customized schedules. This branch is designed only for testers. You should also have gotten an email from me about this.

## How to test
1. Check out this branch and run `bundle exec jekyll serve` as usual.
2. Navigate your browser to `http://localhost:4000/testing/program`.

## Testing notes
- Since ACL 2017 doesn't have a program yet, I used the content from the NAACL 2016 conference for the purposes of testing.
- The page has been tested on MacOS 10.12.2 on recent versions of Chrome, Firefox, and Safari browsers. It would be good to test on other OSes (Windows, Linux) and other browsers, although older IE version (earlier than IE 10) will probably not work.
- The page has also been tested on an iPhone simulator and works as expected.
- Note that if you are using Safari, you will need to use *Command-P* to print and *File > Save as ...* to download the schedule. Chrome and Firefox have buttons for printing and saving as part of their PDF rendering UI.
- Currently, users can only choose papers sessions and individual papers. However, I can certainly extend this to posters since people might want to pick those out in advance too.
- The generated PDF might have some blank rows at the bottom as padding since the PDF generation has been programmed to avoid rows being split across pages.

**Please leave feedback in the form of comments on this PR.**
